File: 1464-maximum-product-of-two-elements-in-an-array/1464-maximum-product-of-two-elements-in-an-array.py
class Solution:
    def maxProduct(self, nums: List[int]) -> int:
        l=len(nums)
        n1,n2=0,0
        for i in nums:
            if (i>n1):
                n2=n1
                n1=i
            elif i>n2:
                n2=i
        return (n1-1)*(n2-1)
        

File: 1464-maximum-product-of-two-elements-in-an-array/1464-maximum-product-of-two-elements-in-an-array.c
int maxProduct(int* nums, int numsSize) {
    int swap;
    for (int i=0;i<numsSize-1;i++){
        swap=0;
        for (int j=0;j<numsSize-1-i;j++){
            if (nums[j]>nums[j+1]){
                int t = nums[j];
                nums[j]=nums[j+1];
                nums[j+1]=t;
                swap=1;
            } 
        }
        if (swap==0) break;
    }
    return (nums[numsSize-1]-1)*(nums[numsSize-2]-1);
}